src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -407,6 +407,33 @@ def _get_installation_id():
     return _get_profile().get_installation_id()
 
 
+@decorators.suppress_all_exceptions(fallback_return="")
+def _get_session_id():
+    # As a workaround to get the terminal info as SessionId, this function may not be accurate.
+
+    def get_hash_result(content):
+        import hashlib
+
+        hasher = hashlib.sha256()
+        hasher.update(content.encode('utf-8'))
+        return hasher.hexdigest()
+
+    # Usually, more than one layer of sub-process will be started when excuting a CLI command. While, the create time
+    # of these sub-processes will be very close, usually in several milliseconds. We use 1 second as the threshold here.
+    # When the difference of create time between current process and its parent process is larger than the threshold,
+    # the parent process will be viewed as the terminal process.
+    import psutil
+    time_threshold = 1
+    process = psutil.Process()
+    while process and process.ppid() and process.pid != process.ppid():
+        parent_process = process.parent()
+        if parent_process and process.create_time() - parent_process.create_time() > time_threshold:
+            content = '{}{}{}'.format(_get_installation_id(), parent_process.create_time(), parent_process.pid)
```

Comment:
I thought you mentioned will distinguish by terminal type right?

Reply:
It turns out to be an invalid method. For the may be the cases where a shell was called in another shell, e.g, powershell will call cmd to execute CLI command. So we can't recognize the target terminal processs only by the name.

I think the method I proposed here is better for practice. It can deal with all kinds of OSes and their different terminals.

Reply:
For automation script, the sessionIDs of the commands in script will be the same except for the first command. It only occurs on None-windwos OSes and only in automation scripts.

Reply:
Okay, you convince me. 👍